### PR TITLE
#164046252 Users can Rate Articles

### PR DESCRIPTION
**Pull request by RachelleMaina**
### Description
Adds Article Rating functionality. Leaving a rating gives authors a general idea of what one feels about their content.

### Type of change
- [x] Non-breaking change (Users can rate each other's articles).

### How Has This Been Tested
- [x] Integration tests
- [x] End to end testing

### Checklist
- [x] Feature passes team's flake8 standards
- [x] New and existing unit tests pass locally with my changes

### How can this be manually tested?
The feature has three endpoints. `GET` is available to any user wheares `PUT` and `POST` are available only to authenticated users.

`POST /api/articles/<slug>/rate/`

Example request body:

```source-json
{
    "rating":{
        "value":"3",
        "review":"Quite interesting."
    }
}
```

`PUT api/articles/<slug>/rate/`

Example request body:

```source-json
{
    "rating":{
        "value":"4",
        "review":"Quite interesting. I loved the ending."
    }
}
```

`GET api/articles/<slug>/ratings/`

#### PT
[#164046248](https://www.pivotaltracker.com/story/show/164046252)

### CODE REVIEW

**Review by Etomovich — APPROVED** [review]

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -129,3 +137,31 @@ class Favorite(models.Model):
         related_name='favorites')
     article_id = models.ForeignKey(
         Article, on_delete=models.CASCADE, related_name='favorites')
+
+
+class Rating(models.Model):
```
Comment (BrianSerem) [inline authors/apps/articles/models.py:142]: Please include a string representation of this class, if there is no string in the body, return a string concatenated with the stringified version of this class's pk, for example: "this is rating str(self.id)"

File: authors/apps/articles/tests/test_article_rating.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+from rest_framework import status
+
+from .article_tests_base_class import ArticlesBaseTest
+from .test_data import *
```
Comment (MandelaK) [inline authors/apps/articles/tests/test_article_rating.py:4]: Please try and import only what is needed, or import the `test_data` module itself

**Review by ElMonstro — APPROVED** [review]
Satisfactory

File: authors/apps/articles/tests/test_data.py [hunk 3]
```diff
@@ -27,3 +27,53 @@
...
+    "message": "You have already rated this article."
+}
+
+update_rating_data_response = {
+    "message": "Rating updated."
+}
+
+not_rated_data_response ={
+"message": "You have not rated this article."
+}
```
Comment (MandelaK) [inline authors/apps/articles/tests/test_data.py]: Please add EOF here

**Review by MandelaK — CHANGES_REQUESTED** [review]

**Comment by RachelleMaina** [comment]
@MandelaK  and @BrianSerem I have made these changes.

**Review by MandelaK — APPROVED** [review]

File: authors/apps/articles/tests/test_article_rating.py (hunk 2 above)
Comment (RachelleMaina) [reply authors/apps/articles/tests/test_article_rating.py:4]: All the data is needed, except one variable. I was merely using the existing convention in the codebase.

File: authors/apps/articles/serializers.py [hunk 4]
```diff
@@ -2,17 +2,20 @@
 
 from authors.apps.profiles.serializers import ProfileSerializer
 
-from .models import Article, Favorite, Like, Snapshot, ThreadedComment
+from .models import Article, Favorite, Like, Snapshot,\
+    ThreadedComment, Rating
```
Comment (cob04) [inline authors/apps/articles/serializers.py]: would you use parenthesis here instead?
```suggestion
  from .models import (Article, Favorite, Like, Snapshot,
    ThreadedComment, Rating)
```

**Review by cob04 — APPROVED** [review]
LGTM

File: authors/apps/articles/serializers.py (hunk 4 above)
Comment (RachelleMaina) [reply authors/apps/articles/serializers.py]: I have corrected this.

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**